lib/matplotlib/cbook/__init__.py

```diff
@@ -1326,7 +1326,11 @@ def _check_1d(x):
         return np.atleast_1d(x)
     else:
         try:
-            x[:, None]
+            shape = x[:, None].shape
+            # work around https://github.com/pandas-dev/pandas/issues/27775
+            # which mean the shape is not as expected
+            if len(shape) != 2:
```

Comment:
that's x[:, None].ndim

Reply:
I would rather check the shape as that is what needs to be correct and do not want to risk `ndim` also being out of sync.

Reply:
If you think than .ndim may be returning something wrong then .shape could also be wrong, no?

Reply:
In the pandas issue they pointed out that this changes was due to an optimization (see https://github.com/pandas-dev/pandas/issues/27775#issuecomment-518819438) on `Index` and the current implementation of `ndim` is:

```python
@property
def ndim(self):
    """
    Number of dimensions of the underlying data, by definition 1.
    """
    return 1

```
https://github.com/pandas-dev/pandas/blob/8b6942f2876264b745529a3c3a7cf410ee096c34/pandas/core/base.py#L694-L699  which confirmed my suspicioun that `shape` and `ndim` were out of sync.  

However, digging through this convinced me that we I should check that `x[:, None].ndim >= 2` as this will fail on older version of pandas and stop relying on an unintentional quirk of Index classes _almost_ ducktyping as arrays.